utilities/backup/backup_engine.cc

```diff
@@ -503,6 +504,14 @@ class BackupEngineImpl {
                                    BackupInfo* backup_info,
                                    bool include_file_details) const;
 
+  // Infers set of existing destination files that could be retained
+  // during the incremental restore procedure in accordance with user
+  // selected strategy (RestoreMode). File can be retained only if it's
+  // deemed to exist in the referenced backup set.
+  void InferDBFilesToRetainInRestore(
+      const std::unique_ptr<BackupMeta>& backup, const std::string& dir,
```

Comment:
Can't `backup` just be `const BackupMeta&`?  (Though this parameter might change entirely based on another comment.)